bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2327,9 +2345,9 @@ public static Image win32_new(Device device, int type, long handle) {
  * @noreference This method is not intended to be referenced by clients.
  */
 public static Image win32_new(Image image, int targetZoom) {
-	if (targetZoom != image.getZoom()) {
-		image.handleDPIChange(targetZoom);
+	if (targetZoom != image.getCurrentDeviceZoom()) {
```

Comment:
win32_new looks like a pointless operation now. Is that still use somewhere and does it still makes sense? It doesn't change anything for the image, right?

Reply:
Its called for setImage in all the handleDPI. Doesnt change anything. We can also re set the image instsad in the callback

Reply:
Yes, because this call is now only creating a new handle internally. I think that method is not needed with your changes anymore. Resetting it in the callbacks should be enough then